src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,141 @@
+                head = newNode;
+            }
+        } else if (index == size) {
+            add(value);
+        } else {
+            Node<T> current = findByIndex(index);
+            current.prev.next = newNode;
+            newNode.prev = current.prev;
+            newNode.next = current;
+            current.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = findByIndex(index);
+        T changedValue = currentNode.item;
+        currentNode.item = value;
+        return changedValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = findByIndex(index);
+        currentNode.prev = currentNode.next;
+        currentNode.next.prev = currentNode.prev;
+        size--;
+        return currentNode.item;
     }
 
     @Override
     public boolean remove(T object) {
+        for (int i = 0; i < size; i++) {
+            Node<T> nodeToRemove = findByIndex(i);
+            if ((object == null && nodeToRemove.item == null)
+                    || (object != null && object.equals(nodeToRemove.item))) {
+                nodeToRemove.prev = nodeToRemove.next;
+                nodeToRemove.next.prev = nodeToRemove.prev;
+                size--;
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size <= 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " doesn't found");
+        }
+    }
+
+    private void checkIndexForAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index " + index + " doesn't found");
+        }
+    }
+
+    private Node<T> findByIndex(int index) {
+        Node<T> currentByIndex;
+        currentByIndex = head;
+        if (index == 0) {
+            return currentByIndex;
+        } else {
+            for (int i = 1; i < size; i++) {
+                currentByIndex = currentByIndex.next;
+                if (i == index) {
+                    return currentByIndex;
+                }
+            }
+        }
+        return currentByIndex;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T element, Node<T> next) {
```

Comment:
Don't use getters and setters in class Node. Since `Node` is an inner class, you have direct access to its private fields.